literature_search.py

```diff
@@ -0,0 +1,298 @@
+            (paper['MedlineCitation']['Article']['ArticleTitle'],
+             paper['MedlineCitation']['PMID'])
+        
+    auths = []
+    try:
+        for auth in paper['MedlineCitation']['Article']['AuthorList']:
+            try:
+                auth_name = [auth['LastName'],auth['Initials']+',']
+                auth_name = ' '.join(auth_name)
+                auths.append(auth_name)
+            except:
+                auths.append('')
+                print(paper['MedlineCitation']['Article']['ArticleTitle'],
+                      'has an issue with an author name')
+    except:
+        auths.append('AUTHOR NAMES ERROR')
+        print(paper['MedlineCitation']['Article']['ArticleTitle'],'has no author list?')
+    authors[i] = ' '.join(auths)
+    journals[i] = '*%s*' % (paper['MedlineCitation']['Article']['Journal']['Title']) 
+    # store keywords 
+    if paper['MedlineCitation']['KeywordList'] != []:
+        kwds = []
+        for kw in paper['MedlineCitation']['KeywordList'][0]:
+            kwds.append(kw[:])         
+        keywords[i] = ' '.join(kwds)
+        
+#==============================================================================
+#========================= Clean up title and word strings ====================
+
+titles = [t.lower() for t in titles] #same case
+titles = [t.replace('<sub>',' ').replace('</sub>','') for t in titles] #subscript
+titles = [t.replace('<i>',' ').replace('</i>','') for t in titles] #italics
+titles = [t.replace('[','').replace(']','') for t in titles] #remove brackets from html parser
+#clean up keywords
+keywords = [k.lower() for k in keywords] #same case
+
+#==============================================================================
+#========================= Loading the things =================================
+
+# Load Top-performing Model
+from keras.models import load_model
+model = load_model('Models/Keras_model/model_DNN.h5')
+print('\nLoaded model from disk')
+
+# Load Associated Vectorizer
+from sklearn.externals import joblib #pickle.load throws warning.
+from sklearn.preprocessing import LabelEncoder
+#load vectorizer and label encoder
+vect = joblib.load(open('Models/Keras_model/Vectorizer_tdif.pkl','rb'))
+le = LabelEncoder()
+le.classes_   = np.load('Models/Keras_model/LabelEncoder.npy')
+print('\nLoaded Vectorizer')
+
+#==============================================================================
+#========================= Vectorize Strings ==================================
+#get titles for this week's literature update
+import pandas as pd
+papers_df = pd.DataFrame({'title': titles,
+                          'keywords': keywords,
+                          'abstract': abstracts,
+                          'author': authors,
+                          'journal': journals})
+
+for index, row in papers_df.iterrows():
+    if row['abstract'] == '' or row['author'] == 'AUTHOR NAMES ERROR' or row['title'] == '':
```

Comment:
Blank abstract and AUTHOR ERROR can exist, but I don't think blank title is a possibility. Can that argument be satisfied? I think papers without a title are not even included, so you might be able to remove ``or row['title'] == ''``. Double check though.

Reply:
It's a good check, just because it hasn't happened yet doesn't mean it won't.

Reply:
There are a million things that haven't happened yet that could happen. Just pointing out that I don't think there's an explicit argument that would set title to blank. Wasn't sure if that was intentional or not.

Reply:
It was, intentional.